src/client/app/actions/groups.ts

```
@@ -316,14 +272,17 @@ function submitGroupEdits(group) {
  * @return {function(*, *)}
  */
 export function submitGroupInEditingIfNeeded() {
-	return (dispatch, getState) => {
+	return (dispatch: Dispatch, getState: GetState) => {
 		if (shouldSubmitGroupInEditing(getState())) {
-			const rawGroup = getState().groups.groupInEditing;
+			const rawGroup = getState().groups.groupInEditing as t.GroupDefinition;
+			if (rawGroup === undefined) {
+				throw new Error('Unacceptable condition: state.groups.groupInEditing has no data.');
```

I would like to remove all the `throw new Error` statements on the client, or at the very least display an error to the user. I don't think they serve much purpose except for debugging purposes, which should not be in production.

In this case, I would just do:
```
if (rawGroups) {
   ...
}
```

---

I highly disagree with you. We shouldn't support our application when it's in weird bad states. Failing fast is always better as it's easier to debug (and at some point we will need to debug something in production), and as the user is just going to have the application subtly break on them later when the application accesses `undefined.startTimestamp` or something. Catching and doing nothing doesn't fix the problem, it just hides it, and it does a bad job of it.

---

And for what it's worth, unless this has been fundamentally changed for typescript, the application will not recover from an empty group in editing marked dirty, and that's the one case that will lead to this error condition. The app will stop working properly whether or not we throw an error.